### DISCUSSION TITLE
How to get as an ouput an ipynb document with answers without code (not just with echo: false)

### DISCUSSION THREAD

**Discussion by finkelshtein** [quarto-dev/quarto-cli#7127] [Q&A]
### Description

Hi,
I would like to have as an output a ipynb file to let students work with it, and I also want to let them compare the answer they obtain with the correct one. However, I want to hide the code from them.
It obviously works for HTML output
````
---
title: "Test"
format: 
  html:
    keep-ipynb: true
jupyter: python3
---

Calculate the product of 2 and 2.

```{python}
# Write your code here
```

Check the answer:
```{python}
#| echo: false
2 * 2
```

````

However, the kept ipynb (naturally) contains the code of the last cell. What I want is to get in the results ipynb, **instead** of the last cell, a markdown cell with just 4 there. 
Does there exist a way to do this?

**Comment by mcanouil** [comment]
Use conditional content, project profile, and directly target the Jupyter Notebook format if you don't want HTML `format: ipynb`.

- https://quarto.org/docs/authoring/conditional.html
- https://quarto.org/docs/projects/profiles.html

<img width="1624" alt="image" src="https://github.com/quarto-dev/quarto-cli/assets/8896044/64b945c1-6fff-4f75-ab0c-137522732376">

**Reply by finkelshtein** [reply]
Thank you for pointing out to the format `ipynb`!
However, I do not want to hide all content, I want to get the output as a Markdown cell.

````
---
title: "Test"
format: ipynb
jupyter: python3
---

Calculate 2+2
```{python}
#| echo: false
2+2
```

````
indeed produces such ipynb-document

![image](https://github.com/quarto-dev/quarto-cli/assets/35048241/76575cda-ee65-4414-9ad1-a19d1f390a2d)

However, that 4 is "fragile": if a student runs the empty cell, 4 disappears. I would like to have **instead** of the empty cell a markdown cell with 4.

Kind regards,
Dmitri

**Reply by mcanouil** [reply]
Then output markdown in a cell with the 4.

**Reply by finkelshtein** [reply]
Unless I misunderstood you - I would like to have this automatically, not manually.

Imagine a document which contains some Python code visible for students and also many exercises with numerical answers. 
For each exercise, I provide a solution which includes a cell with Python code, the last line of the code produces the answer. 
If I put `#| echo: false` in these cells and choose HTML output, students will see the answers and will be able to compare them with their own ones. However, if I want to provide Jupyter Notebook output (e.g. to let students "play" with the visible code), I would need to create manually markdown cells, copy answers from the output, put them into the markdown cells - it's quite boring. (Or did as shown above, but then, as I said, the empty cells will be fragile, they disappear after any rerun)

**Reply by mcanouil** [reply]
I am not sure to understand what you want but I believe you have to write your code cell to do it all, i.e., produce other code cells.

**Reply by finkelshtein** [reply]
Can you advise a reference on how to do this (in Quarto)? Or probably on how to get the output of a cell, e.g. assign it to a variable?
